### PR TITLE
Tag Parameters v0.4.0 v0.5.0

### PR DESCRIPTION
**Pull request by mauro3**
This used to be #6381 but I could not re-open that one.  I think this is correct but I had to edit some stuff by hand.

### CODE REVIEW

File: Parameters/versions/0.5.0/requires [hunk 1]
```diff
@@ -0,0 +1,3 @@
+julia 0.4
+DataStructures
+Compat
```
Comment (tkelman) [inline Parameters/versions/0.5.0/requires]: best to declare a minimum version dependency that implements the features you need

Comment (mauro3) [inline Parameters/versions/0.5.0/requires]: Tnx. I did some detective work and updated https://github.com/mauro3/Parameters.jl/blob/master/REQUIRE and the `requires` for 0.5.  I can't easily update the `REQUIRE` for 0.4 but should I update `Parameters/versions/0.4.0/requires` nonetheless?

Comment (tkelman) [inline Parameters/versions/0.5.0/requires]: if the min ver applies, yes best to edit the metadata requires file

Comment (mauro3) [inline Parameters/versions/0.5.0/requires]: Somewhat related.  In Travis it doesn't actually install the minimal required packages, correct?
See https://travis-ci.org/mauro3/Parameters.jl/jobs/162817273 .  Is there a way to do this?

Comment (tkelman) [inline Parameters/versions/0.5.0/requires]: what do you mean? it installs your dependencies, looks like it's doing the right thing

Comment (mauro3) [inline Parameters/versions/0.5.0/requires]: Yes, but it did not install Compat 0.7.15 but the latest 0.9.2.

Comment (tkelman) [inline Parameters/versions/0.5.0/requires]: oh i see. Pkg tries to maximize versions, you have to pin or specifically ask it otherwise

Comment (mauro3) [inline Parameters/versions/0.5.0/requires]: I guess that makes sense usually. But for testing a switch to checkout the minimal requirements would be good too.

Comment (mauro3) [inline Parameters/versions/0.5.0/requires]: The other `requires` is also adjusted.  Probably ready to merge now.

**Pull request merged**